### ISSUE TITLE
Create SASS utility to calculate accessible text color

### ISSUE THREAD

**Issue by timroes** [airbytehq/airbyte#18155]
We often have the use-case that we have components that render text on a specific background color. We currently need during review check color contrast for those manually to make sure they fulfill accessible requirements, e.g. see https://github.com/airbytehq/airbyte/pull/17479/files#r999221155. We should instead add a SASS utility function `textColorOn`, that will take the background color as a parameter and calculates based on the WCAG2.1 contrast algorithm, whether white or black is the color with the higher contrast and return this, so that we can in the future just write:

```scss
.foo {
  background: colors.$blue;
  color: textColorOn(colors.$blue);
}
```

The utility should ideally also throw an error during build if we're trying to use a background color where neither black nor white fulfill WCAG2.1 AA requirements of contrast ratio.

**Comment by octavia-squidington-iv** [comment]
cc @airbytehq/frontend

**Comment by josephkmh** [comment]
@timroes I don't think this type of SASS utility will be possible with CSS custom properties as our color tokens. At build time SASS will be unable to evaluate color variables and compare contrast. IMO accessible contrast needs to be enforced at the design stage.

**Comment by timroes** [comment]
Yeah, that seems to be a shortcome of having switched to CSS custom properties. There are some ways we could do the calculation in CSS custom properties (see https://css-tricks.com/switch-font-color-for-different-backgrounds-with-css/), but that would require us again to have a way of turning a hex color into it's R, G and B component, which we can't :-/

There is only one other way I'd see, which is we let SASS calculate the text colors at build time in the theme file, e.g. something like, but that would require us to be able to get all color variable names in that file (which we can't) or duplicate all color names again (which I don't think is worth it), i.e. something like:

```scss
:root {
  // ...
}
```
